blog/content/post/2024-04-12-odroid-m1-secure-boot.md

```diff
@@ -0,0 +1,598 @@
+couple variables:
+
+```shell
+export CROSS_COMPILE=aarch64-linux-gnu-
+export BL31=<path/to/rkbin>/bin/rk35/rk3568_bl31_v1.44.elf
+export ROCKCHIP_TPL=<path/to/rkbin>/bin/rk35/rk3568_ddr_1560MHz_v1.21.bin
+```
+
+This time I used `gcc-aarch64-linux-gnu` cross-compiler from Debian package
+manager.
+
+### Build
+
+After configuration, we build by using `make`. It should build signed U-Boot with
+public key embedded inside SPL.
+
+```text
+make -j$(nproc)
+(...)
+ Default Configuration: 'config-1'
+ Configuration 0 (config-1)
+  Description:  rk3568-odroid-m1.dtb
+  Kernel:       unavailable
+  Firmware:     atf-1
+  FDT:          fdt-1
+  Loadables:    u-boot
+                atf-2
+                atf-3
+                atf-4
+                atf-5
+                atf-6
+  Sign algo:    sha256,rsa2048:dev
+  Sign value:   c05e2589eb863bfb9b5b5c87ea9f99042c4cb78797a6cd0d772a68bfac21bc621790cb977bd414928b7b88a50273d336aea4afb7a1a4008834e1c3c18eb08fb617bb71205e6773904f42e81364a15eb5c581425f22d6be5e30dd1a44cbe7fa626b7fb34a3eb742c03b0ec47c19dd957358949764f82ffb24f9359efe083ea04ada512f8cab0f469cac28d08da3dfffbde1516cb225b3011f36495c959793d11b8c4bde234caca5ac1d779f7983dcf865be92c2ac3300e4cdf536e51fa398c8930dafbdd0a3258ba4704eebc063e5e57533a962a2da5c7eaf447b1244f9720adb3dc775d95b4c6dc99f6b9bd73ebe992095429510b86dd5ac912c24a8cec64841
+  Timestamp:    Mon May 27 15:08:11 2024
+Signature written to 'u-boot.itb', node '/configurations/config-1/signature'
+  OFCHK   .config
+```
+
+### Signing idbloader
+
+Signing idbloader is similar to [Signing Loader](#signing-loader) section except
+with `sb --idb` argument.
+It's important to remember to sign idbloader with the same keys as used in that
+section (in case current ones are different).
+
+```shell
+tools/rk_sign_tool sb --idb ../u-boot/idbloader.img
+********sign_tool ver 1.4********
+IDB binary is ../u-boot/idbloader.img
+signing idbhead...
+failed to get key = sign_algo
+signing idbloader ok
+```
+
+### U-Boot Verification
+
+To check if SPL is signed correctly and that it correctly verifies U-Boot I have
+written idbloader.img file to `spl` partition created in
+[Writing key hash to OTP](#writing-key-hash-to-otp). I also flashed `u-boot.itb`
+file to `uboot` partition.
+
+After writing needed files and inserting SD card into ODROID i restarted
+platform while keeping recovery button pressed.
+
+```text
+U-Boot SPL 2024.01-dirty (Apr 11 2024 - 09:15:28 +0200)
+Trying to boot from MMC2
+## Checking hash(es) for config config-1 ... sha256,rsa2048:dev+ OK
+## Checking hash(es) for Image atf-1 ... sha256+ OK
+## Checking hash(es) for Image u-boot ... sha256+ OK
+## Checking hash(es) for Image fdt-1 ... sha256+ OK
+## Checking hash(es) for Image atf-2 ... sha256+ OK
+## Checking hash(es) for Image atf-3 ... sha256+ OK
+## Checking hash(es) for Image atf-4 ... sha256+ OK
+## Checking hash(es) for Image atf-5 ... sha256+ OK
+## Checking hash(es) for Image atf-6 ... sha256+ OK
+(...)
+=>
+```
+
```

Comment:
You could live here a sentence to sum up this log or to state, whether your expectations were met or not and how you made the conclusion. For example, I do not understand whether this attempt is successful or not, and why.

Reply:
f7982c42f7d027003915f05f61168e15a1a64921